### PR TITLE
Feat: add "placeholder" support in Markdown and WYSIWYG mode (fix #117)

### LINKED ISSUE #117: Need PlaceHolder for the Editor
It would be nice feature to have the Place Holder option in the Editor

### PR DESCRIPTION
**Pull request by reyronald**
### Please check if the PR fulfills these requirements

- [ ] It's the right issue type on the title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has a description of the breaking change

### Description

Add support for "placeholder" option in WYSIWYG mode.

Fix #117 

placeholder

---

Before I spend time adding tests, updating documentation, JSDoc, doing more proper manual regressions tests, etc. I would like to know if this a contribution that you would accept and are willing to review and work together with me to get done.

I volunteer some of my own time to implement this properly since I could use this feature and would prefer it to be supported natively than hacking it on my own in my consuming application.

If the answer is positive, then I'd like to know how you feel about this approach, if the location of the code aligns with the rest of the structure and architecture, and of course any general suggestions you might have.

And by the way, this is a great and solid editor! Congratulations to you and whole team of contributors for working on it and putting it out there for the community to use! You have my gratitude! 🎉 👍

### CODE REVIEW

File: src/css/tui-editor-contents.css [hunk 1]
```diff
@@ -237,3 +237,9 @@
     margin-right: 3.8px;
     margin-top: 3px;
 }
+
+.tui-editor-contents-placeholder:not(:focus):before {
+    content: attr(data-placeholder);
+    color:grey;
+    font-style: italic;
+}
```
Comment (sohee-lee7) [inline src/css/tui-editor-contents.css]: Please insert new line

File: src/js/editor.js [hunk 2]
```diff
@@ -217,6 +217,8 @@ class ToastUIEditor {
 
     this.setValue(this.options.initialValue, false);
 
+    this.setPlaceholder(this.options.placeholder);
```
Comment (sohee-lee7) [inline src/js/editor.js]: Please add a comment to the above JS Doc comment above for `placeholder`. The proper position is 130 line that is below the comment of `customConvertor`.

File: src/js/editor.js [hunk 3]
```diff
@@ -771,6 +773,10 @@ class ToastUIEditor {
     return textObject.getTextContent() || '';
   }
 
+  setPlaceholder(placeholder) {
```
Comment (sohee-lee7) [inline src/js/editor.js]: Please add a comment for JS Doc.

File: src/js/wysiwygEditor.js [hunk 4]
```diff
@@ -434,6 +437,14 @@ class WysiwygEditor {
     });
   }
 
+  _togglePlaceholder() {
+    if (this.get$Body()[0].textContent === '') {
```
Comment (sohee-lee7) [inline src/js/wysiwygEditor.js]: Please change condition. 
```js
if (this.get$Body()[0].textContent)
```
This is our coding convention.

Also do not call `this.get$Body()` every time in this function, but assign it to a variable in this function like this:
```js
const $body = this.get$Body();
if ($body[0].textContent) {
    $body.addClass('tui-editor-contents-placeholder');
} else {
    $body.removeClass('tui-editor-contents-placeholder');
}
```

**Comment by reyronald** [comment]
Hey @sohee-lee7 	,

Take another look now, I addressed your suggestions and also implemented the placeholder feature on Markdown mode.

I tested this in Chrome, Firefox and Edge and worked as expected in all cases.

Let me know what you think.

File: src/css/tui-editor-contents.css [hunk 5]
```diff
@@ -237,3 +237,11 @@
     margin-right: 3.8px;
     margin-top: 3px;
 }
+
+.tui-editor-contents-placeholder:not(:focus):before {
+    content: attr(data-placeholder);
+    color:grey;
+    font-style: italic;
```
Comment (sohee-lee7) [inline src/css/tui-editor-contents.css]: Placeholder of markdown is not italic so I think this font style should be erased.

**Comment by sohee-lee7** [comment]
@reyronald Thank you. I'll merge soon! 👍

**Comment by sohee-lee7** [comment]
@reyronald The release date has not yet been set. I think it will release in a week or two.

**Comment by chkaos** [comment]
Hi, can you tell me how to enable the placeholder?

**Pull request merged**